crates/cargo-util/src/process_error.rs

```diff
@@ -61,46 +85,26 @@ impl ProcessError {
     ///
     /// * `code` can be `None` for situations like being killed by a signal on unix.
     pub fn new_raw(
-        msg: &str,
+        msg: impl Into<String>,
         code: Option<i32>,
-        status: &str,
-        stdout: Option<&[u8]>,
-        stderr: Option<&[u8]>,
+        status: impl Into<String>,
+        stdout: Option<impl Into<Vec<u8>>>,
+        stderr: Option<impl Into<Vec<u8>>>,
     ) -> ProcessError {
-        let mut desc = format!("{} ({})", msg, status);
-
-        if let Some(out) = stdout {
-            match str::from_utf8(out) {
-                Ok(s) if !s.trim().is_empty() => {
-                    desc.push_str("\n--- stdout\n");
-                    desc.push_str(s);
-                }
-                Ok(..) | Err(..) => {}
-            }
-        }
-        if let Some(out) = stderr {
-            match str::from_utf8(out) {
-                Ok(s) if !s.trim().is_empty() => {
-                    desc.push_str("\n--- stderr\n");
-                    desc.push_str(s);
-                }
-                Ok(..) | Err(..) => {}
-            }
-        }
-
         ProcessError {
-            desc,
+            msg: msg.into(),
             code,
-            stdout: stdout.map(|s| s.to_vec()),
-            stderr: stderr.map(|s| s.to_vec()),
+            status: status.into(),
+            stdout: stdout.map(Into::into),
+            stderr: stderr.map(Into::into),
         }
     }
 
     /// Creates a [`ProcessError`] with "could not execute process {cmd}".
     ///
     /// * `cmd` is usually but not limited to [`std::process::Command`].
     pub fn could_not_execute(cmd: impl fmt::Display) -> ProcessError {
-        ProcessError::new(&format!("could not execute process {cmd}"), None, None)
```

Comment:
Please focus commits on a single change, rather than mixing changes in them, obscuring what you are doing.